### PR TITLE
dns managed zone with plugin-framework

### PR DESCRIPTION
**Pull request by megan07**
This will be merged after the framework is reviewed and merged, but fixes https://github.com/hashicorp/terraform-provider-google/issues/12911

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-the-terraform-providers), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/main/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:none

```

### CODE REVIEW

**Comment by SarahFrench** [comment]
I'll review data_source_dns_managed_zone.go and data_source_dns_managed_zone_test.go, as I assume there are no other difference between this PR and the other PR that was merged into main. But if there are, please give me a heads up!

File: mmv1/third_party/terraform/tests/data_source_dns_managed_zone_test.go.erb [hunk 1]
```diff
@@ -15,24 +16,32 @@ func TestAccDataSourceDnsManagedZone_basic(t *testing.T) {
...
+			},
+		},
+		CheckDestroy: testAccCheckDNSManagedZoneDestroyProducerFramework(t),
 		Steps: []resource.TestStep{
 			{
 				Config: testAccDataSourceDnsManagedZone_basic(randString(t, 10)),
 				Check: checkDataSourceStateMatchesResourceStateWithIgnores(
 					"data.google_dns_managed_zone.qa",
 					"google_dns_managed_zone.foo",
 					map[string]struct{}{
```
Comment (SarahFrench) [inline mmv1/third_party/terraform/tests/data_source_dns_managed_zone_test.go.erb:32]: The acceptance test changes that compare the new and old providers were eaten by rebasing 😢  - they're needed to approve this PR

File: mmv1/third_party/terraform/data_sources/data_source_dns_managed_zone.go [hunk 2]
```diff
@@ -1,101 +1,151 @@
...
-				Type:     schema.TypeString,
+		Attributes: map[string]schema.Attribute{
+			"dns_name": schema.StringAttribute{
 				Computed: true,
 			},
 
-			"name": {
-				Type:     schema.TypeString,
+			"name": schema.StringAttribute{
 				Required: true,
```
Comment (SarahFrench) [inline mmv1/third_party/terraform/data_sources/data_source_dns_managed_zone.go:54]: Do we need `MarkdownDescription` for the attributes?

I guess not, as the docs for this data source are handwritten, but I think it'd be great to include `MarkdownDescription` from the start of using the plugin framework as it makes any automatic docs generation easier to do in future

**Review by SarahFrench — COMMENTED** [review]
Submitting this review early so you get feedback earlier! I will probably follow up with some more comments

File: mmv1/third_party/terraform/tests/data_source_dns_managed_zone_test.go.erb (hunk 1 above)
Comment (megan07) [reply mmv1/third_party/terraform/tests/data_source_dns_managed_zone_test.go.erb:32]: ah!! 🤦‍♀️  - i figured this was going to happen. my bad, will get those back

Comment (SarahFrench) [reply mmv1/third_party/terraform/tests/data_source_dns_managed_zone_test.go.erb:32]: I checked - [the acc test commit](https://github.com/GoogleCloudPlatform/magic-modules/pull/7182/commits/ae6a45996cb26ac6e1c0dffd0e5a0cf58e900197) also included the test helper functions, which I think you subsequently put in a framework test utils file

File: mmv1/third_party/terraform/tests/data_source_dns_managed_zone_test.go.erb [hunk 3]
```diff
@@ -15,24 +15,50 @@ func TestAccDataSourceDnsManagedZone_basic(t *testing.T) {
...
-						"forwarding_config.#":         {},
-						"force_destroy":               {},
-						"labels.#": 				   {},
-						"creation_time": 			   {},
+						"dnssec_config.#":                       {},
+						"private_visibility_config.#":           {},
+						"peering_config.#":                      {},
+						"forwarding_config.#":                   {},
+						"force_destroy":                         {},
+						"labels.#": 				                     {},
```
Comment (SarahFrench) [inline mmv1/third_party/terraform/tests/data_source_dns_managed_zone_test.go.erb:55]: I'm a bit suspicious about this test as there was the issue before where it turned out labels.# wasn't being used in the test using v4.50.0 (there wasn't a a labels attribute to be ignored in that test) and that line needed to be changed to labels.% to work with the muxed provider (as there was a labels attribute and it had a % not a #). See here : https://github.com/hashicorp/magic-modules/blob/06984badd83acfc915b9e82331bf19156dd7e00b/mmv1/third_party/terraform/tests/data_source_dns_managed_zone_test.go.erb

I don't understand how the vcr tests have passed on this pr given that issue... 🤷‍♀️

**Review by SarahFrench — COMMENTED** [review]
Quick fly-by comment before real review next week

File: mmv1/third_party/terraform/tests/data_source_dns_managed_zone_test.go.erb (hunk 1 above)
Comment (SarahFrench) [reply mmv1/third_party/terraform/tests/data_source_dns_managed_zone_test.go.erb]: I'm not sure if it's strictly needed, but the original acceptance test PR into this PR included comparing values between the two test runs (see use of `dns_name1` etc[ in the diff here](https://github.com/hashicorp/magic-modules/pull/7/files)). Do you think this helps improve the test? For the managed zone resource it's unclear what field(s) could differ.

**Review by SarahFrench — COMMENTED** [review]
I noticed another diff in the acceptance tests which may/may not be added depending on how useful we feel it is? ([See comment in thread](https://github.com/GoogleCloudPlatform/magic-modules/pull/7182#discussion_r1097394017)).
After that's resolved I'm happy to approve!

File: mmv1/third_party/terraform/tests/data_source_dns_managed_zone_test.go.erb (hunk 1 above)
Comment (SarahFrench) [reply mmv1/third_party/terraform/tests/data_source_dns_managed_zone_test.go.erb]: We decided it's not needed for this data source

**Review by SarahFrench — APPROVED** [review]
LGTM!

**Pull request merged**